core/org/commcare/core/sandbox/SandboxUtils.java

```diff
@@ -0,0 +1,130 @@
+    public static Vector<String> extractEntityOwners(UserSandbox sandbox) {
+        Vector<String> owners = new Vector<String>();
+        Vector<String> users = new Vector<String>();
+
+        for (IStorageIterator<User> userIterator = sandbox.getUserStorage().iterate(); userIterator.hasMore(); ) {
+            String id = userIterator.nextRecord().getUniqueId();
+            owners.addElement(id);
+            users.addElement(id);
+        }
+
+        //Now add all of the relevant groups
+        //TODO: Wow. This is.... kind of megasketch
+        for (String userId : users) {
+            DataInstance instance = loadFixture(sandbox, "user-groups", userId);
+            if (instance == null) {
+                continue;
+            }
+            EvaluationContext ec = new EvaluationContext(instance);
+            for (TreeReference ref : ec.expandReference(XPathReference.getPathExpr("/groups/group/@id").getReference())) {
+                AbstractTreeElement<AbstractTreeElement> idelement = ec.resolveReference(ref);
+                if (idelement.getValue() != null) {
+                    owners.addElement(idelement.getValue().uncast().getString());
+                }
+            }
+        }
+
+        return owners;
+    }
+
+    /**
+     * A quick way to request an evaluation context with an abstract instance available.
+     *
+     */
+    public static EvaluationContext getInstanceContexts(UserSandbox sandbox, String instanceId, String instanceRef){
+        InstanceInitializationFactory iif = new CommCareInstanceInitializer(sandbox);
+
+        Hashtable<String, DataInstance> instances = new Hashtable<String, DataInstance>();
+        ExternalDataInstance edi = new ExternalDataInstance(instanceRef, instanceId);
+        edi.initialize(iif, instanceId);
+        instances.put(instanceId, edi);
+
+        return new EvaluationContext(null, instances);
+    }
+
+    /**
+     * Load the referenced fixture out of storage for the provided user
+     *
+     * @param sandbox The current user's sandbox
+     * @param refId The jr:// reference
+     * @param userId The user's ID
+     * @return The form instance matching the refId in the sandbox
+     */
+    public static FormInstance loadFixture(UserSandbox sandbox,
+                                            String refId, String userId) {
+        IStorageUtilityIndexed<FormInstance> userFixtureStorage =
+                sandbox.getUserFixtureStorage();
+        IStorageUtilityIndexed<FormInstance> appFixtureStorage =
+                sandbox.getAppFixtureStorage();
+
+        Vector<Integer> userFixtures =
+                userFixtureStorage.getIDsForValue(FormInstance.META_ID, refId);
+        // ... Nooooot so clean.
+        if (userFixtures.size() == 1) {
+            // easy case, one fixture, use it
+            return userFixtureStorage.read(userFixtures.elementAt(0).intValue());
+            // TODO: Userid check anyway?
+        } else if (userFixtures.size() > 1) {
```

Comment:
Everything below this line in this method isn't covered by the test suite. More of a heads up because I'm guessing Clayton wrote this code.